### ISSUE TITLE
Please make a releaseb

### ISSUE THREAD

**Issue by Chris2011** [drrb/rust-netbeans#17]
It would be great, if we can have a pre-build nbm file here and if it is still usable and releasable please upload it to the plugin portal of NetBeans.

Regards

Chris

**Comment by drrb** [comment]
Working on it! I'm currently updating it to work with newest Rust and NetBeans versions.

**Comment by Chris2011** [comment]
Only to ask, I wanted to build the plugin and it needs ruby, is there a specific reason why? Thx.

**Comment by drrb** [comment]
Yes, it's for generating sources while I'm updating the Rust parser. I'll probably remove it once the parser is more stable. Meanwhile, you should be able to build it without Ruby like this:

```
$ mvn package -Dexec.skip -DskipTests
```

Don't expect too much at the moment. The parser is about 1/3 complete, so you'll probably see exceptions in the UI.

**Comment by Chris2011** [comment]
No problem. Thx :) Will test it later.

**Comment by Chris2011** [comment]
I can't install the plugin, because of this error:

![image](https://user-images.githubusercontent.com/795658/31051034-21ced4ca-a65e-11e7-96fb-91b23b7eaf85.png)

**Comment by drrb** [comment]
We're currently depending on a pretty old implementation version of the Common Test Runner API that came with NetBeans 8.0.x. From memory, the Common Test Runner API required pinning to a specific implementation version so that we could access the API. I'll take another look into it when I update the plugin for more recent NetBeans versions. Opened #18 to track the CTR API issue, and #19 to track porting to newer versions of NetBeans.

**Comment by Chris2011** [comment]
Great, thx :) Would be cool to test it soon. I know that there is not that much inside but, a bit is better than nothing.

**Comment by drrb** [comment]
There's a NetBeans 8.2 branch now (https://github.com/drrb/rust-netbeans/tree/netbeans-8.2). One test is still having problems, but you should be able to build that branch for use with NetBeans 8.2.

**Comment by Chris2011** [comment]
Great thx :)

**Comment by Chris2011** [comment]
Great it works :) Should I create more tickets for that what is not working now or should I wait?

**Comment by drrb** [comment]
Good to hear. You're welcome to create tickets if you like. For things like the parser where there's a lot missing, please keep tickets high-level (e.g. "Some language features not supported by parser", rather than, e.g. "Impls in source files don't work"). I'm happy for you create pull requests too.

**Comment by Chris2011** [comment]
Because of the broken parser, did you created the javacc grammar file, by your own? Maybe antlr can do it better? See this grammar, maybe it will help: https://github.com/jbclements/rust-antlr

Kind of old, but maybe better. Only guessing. I know then you have to switch to ANTLR, I did this for my C# plugin for NetBeans too. Because there was no javacc grammar file for C#.

**Comment by drrb** [comment]
Yes, the grammar is still in progress, because I'm writing it myself. I initially did use jbclements/rust-antlr (in 2013), and then switched to using the native parser from rustc, and now to building a JavaCC parser myself. It's partly because the rust-antlr is quite old and slow to parse large documents, and partly because I'm interested in writing the parser myself to learn. My plan is to gradually support all of Rust using the JavaCC parser.

**Comment by Chris2011** [comment]
ok, thx for the info :)

**Comment by Chris2011** [comment]
Hi, what is still missing to make a first alpha release? I will check back the plugin again. Are you still working on it?

**Comment by Chris2011** [comment]
I would like to catch this project up again, it get's more interesting for me. So can you please help here @drrb and maybe add a discussion section for what needs to be done? What the progress of the parser is. Do we really need it or we switch to textmate lexer and use language server for Rust, etc?
